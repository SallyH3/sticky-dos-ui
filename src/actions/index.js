export const setCardList = cardList =>{
         console.log("VIP List", cardList)
         return {
           type: "SET_CARDLIST",
           cardList
         }};

export const deleteCard = (id) => ({
	type: 'DELETE_CARD',
	id
})