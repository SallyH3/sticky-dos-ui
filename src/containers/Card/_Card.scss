.Card {
  box-sizing: border-box;
  max-height: 60vh;

  h4 {
    margin: none;
  }

  button {
    background: none;
    border: none;
    height: 10px;
  }

    .content {
      i {
        margin-right: 10px;
      }
    }
}

.Card__top {
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  align-items: center;
  position: relative;
  padding: 10px 0 0 25px;
  max-height: 25%;
  width: 100%;
  
  h4 {
    display: flex;
    align-items: center;
    margin: none;
    height: 95%;
    width: 75%;
    overflow-y: scroll;
  }
}

.Card__Btns {
  width: 5%;
  display: flex;
  justify-content: space-between;

    button {
      font-size: 16px;
      padding-bottom: 20px;
    }
}