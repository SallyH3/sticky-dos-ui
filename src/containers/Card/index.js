import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteCard } from '../../actions';
import { Redirect } from 'react-router-dom';

export class Card extends Component {
  constructor() {
    super()
    this.state = {
      redirect: false
    }
  }

  handleClick = () => {
    this.deleteCard(this.props.id)

  }

  deleteCard = (id) => {
    const url = `http://localhost:3001/api/v1/cardList/${id}`
    fetch(url, {
      method: 'DELETE'
    })
    this.props.deleteCard(id)
  };

  mapListItems = (content) => {
    return content.map(li => (
      <p key={li.id} className="listItem">
        {!li.checked && <i className="far fa-square" />}
        {li.checked && <i className="fas fa-check-square" />}
        {li.text}
      </p>
    ));
  }

  

  render() {
    const {title, content, id} = this.props

    if (this.state.redirect) {
      return <Redirect to={`/notes/${id}`} />;
    }

    return (
      <article className="Card">
        <button onClick={()=> this.setState({redirect: true})}>
          <i className="fas fa-edit" />
        </button>
        <button onClick={this.handleClick} className="Card__trash">
          <i className="fas fa-trash" />
        </button>
          <section className="Card__header">
            <h4>{title}</h4>
          </section>
          <div className="content">
            {content[0].type === "note" && <p>{content[0].text}</p>}
            {content[0].type === "list" && this.mapListItems(content)}
          </div>
      </article>
    );
  }
}

export const mapDispatchToProps =(dispatch) => ({
  deleteCard: (id) => dispatch(deleteCard(id))
})

export default connect(null, mapDispatchToProps)(Card)