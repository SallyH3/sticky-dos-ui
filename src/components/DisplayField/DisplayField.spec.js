import React from 'react';
import { shallow } from 'enzyme';
import { DisplayField, mapStateToProps } from './index.js';
import Card from '../../containers/Card';
import { cardList } from '../../actions';

describe('DisplayField', () => {
  let wrapper;

  beforeEach(() => {
    wrapper=shallow(
      <DisplayField />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})