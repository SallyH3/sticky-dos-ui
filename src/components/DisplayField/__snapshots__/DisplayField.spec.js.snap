// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayField should match snapshot 1`] = `undefined`;
